### PR TITLE
Add relations migrator command 

### PR DESCRIPTION
**Pull request by coderbydesign**
## Description of Intent of Change(s)
[Add management command for running relations migrator](https://github.com/RedHatInsights/insights-rbac/commit/b32d75dd859f7968c48d31af748c8caab039f66c) 

Adds the ability to run the relations migrator code (with params to commit, exclude
apps and supply an org list) via the Django CLI for ease of testing locally.

See README updates in commit for instructions.

[Remove superfluous error handler and update calls](https://github.com/RedHatInsights/insights-rbac/commit/681875848735ca729b1b5813e146373590244e9d) 

This updates the error logging/raising for gRPC calls by accessing the `code()` and
`details()` methods on `grpc.RpcError` directly.

## Local Testing
Follow the instructions on the updated README to run the migrator command.

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: rbac/migration_tool/utils.py [hunk 1]
```diff
@@ -74,9 +49,8 @@ def write_relationships(relationships):
         try:
             stub.CreateTuples(request)
         except grpc.RpcError as err:
-            error = GRPCError(err.value)
```
Comment (astrozzc) [inline rbac/migration_tool/utils.py:77]: Will the GRPCError(err) work? We got the GRPCError from QE, thought it might be useful to construct the error

**Review by astrozzc — APPROVED** [review]

**Comment by lpichler** [comment]
Now it looks CI failure is related to PR:
from https://ci.ext.devshift.net/job/RedHatInsights-insights-rbac-pr-check/2352/consoleFull :

```
 Successfully installed cffi-1.17.0 cryptography-43.0.0 pycparser-2.22 types-cffi-1.16.0.20240331 types-pyOpenSSL-24.1.0.20240722 types-python-dateutil-2.9.0.20240316 types-pytz-2024.1.0.20240417 types-redis-4.6.0.20240806 types-requests-2.32.0.20240712 types-setuptools-71.1.0.20240813 urllib3-2.2.2
08:52:49 rbac/management/management/commands/migrate_relations.py:5: error: Module "migration_tool.migrate" has no attribute "migrate_data"  [attr-defined]
08:52:49 Installing missing stub packages:
```

File: rbac/migration_tool/utils.py (hunk 1 above)
Comment (lpichler) [reply rbac/migration_tool/utils.py:77]: as I remember I think that in `error.reason` is more details - all errors are not is in err.details() -  @vbelchio  is it correct ?

**Comment by lpichler** [comment]
> Now it looks CI failure is related to PR: from https://ci.ext.devshift.net/job/RedHatInsights-insights-rbac-pr-check/2352/consoleFull :
> 
> ```
>  Successfully installed cffi-1.17.0 cryptography-43.0.0 pycparser-2.22 types-cffi-1.16.0.20240331 types-pyOpenSSL-24.1.0.20240722 types-python-dateutil-2.9.0.20240316 types-pytz-2024.1.0.20240417 types-redis-4.6.0.20240806 types-requests-2.32.0.20240712 types-setuptools-71.1.0.20240813 urllib3-2.2.2
> 08:52:49 rbac/management/management/commands/migrate_relations.py:5: error: Module "migration_tool.migrate" has no attribute "migrate_data"  [attr-defined]
> 08:52:49 Installing missing stub packages:
> ```

Oh it I guess it needs Jays PR.

File: rbac/migration_tool/utils.py (hunk 1 above)
Comment (coderbydesign) [reply rbac/migration_tool/utils.py:77]: I'll pull this commit out for now. I was getting errors here when the connection wasn't successful, but it's outside the scope of this anyway and it sounds like I'm missing some context. :)

File: README.rst [hunk 2]
```diff
@@ -115,6 +115,14 @@ To run the local dev Django on a specific port use::
 
     make PORT=8111 serve
 
+Migrating Relations
+^^^^^^^^^^^^^^^^^^^
+
+To run the migrator tool to convert RBAC data into [Kessel relations](https://github.com/project-kessel/relations-api), use ::
+
+    ./rbac/manage.py migrate_relations [--org-list ORG_LIST [ORG_LIST ...]] [--exclude-apps EXCLUDE_APPS [EXCLUDE_APPS ...]] [--write-to-db]
+
```
[the remark this answers is not in the document]
Comment (coderbydesign) [reply README.rst:124]: Added!

**Comment by coderbydesign** [comment]
@lpichler yeah, it is blocked on #1164 - I just noted that in the title. :)

**Comment by lpichler** [comment]
@coderbydesign can you rebase please 🙏 ?

**Pull request merged**